core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3256,11 +3256,12 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
   }
 
   test("test 1 resource profile") {
+    val sparkContext = sc
```

Comment:
https://github.com/apache/spark/blob/797abc069348a2770742d5b57fd8c0fab0abe8d4/core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala#L302

If there is no problem with the original UT design, I'm not sure if this line should be changed to `nextProfileId.incrementAndGet()`, needs to be investigated again

Reply:
Ignore comments above

Reply:
Good catch. we should set the nextProfileId to 1, since 0 is set to default. Or else, the profile ID will be messed up

Reply:
I will change it accordingly

Reply:
Hmm...this may not be perfect. If we initialize `default ` first, profile with id 1 will no longer exist

Reply:
@LuciferYang Yep, the id 1 will not exist in this case. IMO, the rp id is just an identifier, and the value itself does not have any meaning. So I'm thinking it's okay. And for below scenario,

``` scala
    val sc = new SparkContext("local[12]", "rp")
    sc.stop()
    val sc1 = new SparkContext( "local[12]","rp")
    val ereqs = new ExecutorResourceRequests().cores(4)
    val treqs = new TaskResourceRequests().cpus(1)
    val rp1 = new ResourceProfileBuilder().require(ereqs).require(treqs).build
    println(s"rp1.id = ${rp1.id}")
    sc1.stop()
```

we will get rp1.id=1

the profile with id 1 will also no longer exist.